hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/HandleWorkflow.java

```diff
@@ -240,22 +258,25 @@ private VerificationResult getVerifications(
 
         // If pre-handle was successful, we return the result. Otherwise, we charge the node or throw an exception.
         return switch (result.status()) {
-            case SO_FAR_SO_GOOD -> new VerificationResult(result);
+            case SO_FAR_SO_GOOD -> result;
             case NODE_DUE_DILIGENCE_FAILURE -> createPenaltyPayment();
             case UNKNOWN_FAILURE -> throw new IllegalStateException("Pre-handle failed with unknown failure");
             default -> throw new PreCheckException(result.responseCode());
         };
     }
 
     @NonNull
-    private VerificationResult createPenaltyPayment() {
+    private PreHandleResult createPenaltyPayment() {
         // TODO: Implement createPenaltyPayment() - https://github.com/hashgraph/hedera-services/issues/6811
         throw new UnsupportedOperationException("Not implemented yet");
     }
 
-    private boolean preHandleStillValid(@Nullable final Object metadata) {
-        // TODO: Check config (https://github.com/hashgraph/hedera-services/issues/6812)
-        return metadata instanceof PreHandleResult;
+    private boolean preHandleStillValid(
+            @NonNull final VersionedConfiguration configuration, @Nullable final Object metadata) {
+        if (metadata instanceof PreHandleResult preHandleResult) {
+            return preHandleResult.configVersion() == configuration.getVersion();
+        }
+        return false;
```

Comment:
Maybe we should add a log statement here, because I am not sure it should be possible for the metadata to be null, and certainly there is no chance it should be anything other than a PreHandleResult. I would like to know if that ever happens so we can drill into the cause and determine whether it is legitimate or not.

Reply:
`metadata` can be null if pre-handle has not finished yet. Also, @tinker-michaelj mentioned there are special scenarios when we run handle without running pre-handle first.